### PR TITLE
[Kubernetes][Docs] GPU usage

### PR DESCRIPTION
**Pull request by DmitriGekhtman**
## Why are these changes needed?

This adds a note on GPU usage to the Kubernetes documentation.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/kubernetes.rst [hunk 1]
```diff
@@ -292,6 +297,20 @@ To delete a running Ray cluster, you can run the following command:
 
   kubectl delete -f ray/doc/kubernetes/ray-cluster.yaml
 
+.. _k8s-gpus:
+
+Using GPUs
+----------
+
+Using GPUs on Kubernetes requires some setup.
+Here's where to find the relevant documentation for `GKE`_, for `EKS`_, and for `AKS`_.
```
Comment (richardliaw) [inline doc/source/cluster/kubernetes.rst:306]:
```suggestion
Here's where to find the relevant documentation for GPU usage on `GKE`_, for `EKS`_, and for `AKS`_.
```

File: doc/source/cluster/kubernetes.rst [hunk 2]
```diff
@@ -292,6 +297,20 @@ To delete a running Ray cluster, you can run the following command:
...
+
+The `Ray Docker Hub <https://hub.docker.com/r/rayproject/>`_ hosts CUDA-based images packaged with Ray for use in Kubernetes pods. 
+For example, the image ``rayproject/ray-ml:nightly-gpu`` is ideal for running GPU-based ML workloads with the most recent nightly build of Ray.
+Read :ref:`here<docker-images>` for further details on Ray images. 
+
+If you run into problems setting up GPUs for your Ray cluster on Kubernetes, please reach out to us at `<https://discuss.ray.io>`_. 
+
```
Comment (richardliaw) [inline doc/source/cluster/kubernetes.rst:313]: Can you also mention that you need to explicitly change the resource requirements for the containers in your yaml?

Also, a snippet of relevant changes would be really helpful (which I think is only a couple lines of code)!

**Comment by AmeerHajAli** [comment]
@DmitriGekhtman , Can you please fix lint so Richard can merge this?

**Comment by DmitriGekhtman** [comment]
@AmeerHajAli @richardliaw 
Fixed a minor markup issue. Now there are no errors related to my changes when I run `sphinx-build -q -E -W -T -b html source _build/html` locally, and the docs build above succeeds.

File: doc/source/cluster/kubernetes.rst [hunk 3]
```diff
@@ -292,6 +297,79 @@ To delete a running Ray cluster, you can run the following command:
...
+GPU taints and tolerations
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. note::
+
+  Users using a managed Kubernetes service probably don't need to worry about this section. 
+
+The `Nvidia gpu plugin`_ for Kubernetes applies `taints`_ to GPU nodes; these taints prevent non-GPU pods from being scheduled on GPU nodes.
+Managed Kubernetes services like GKE, EKS, and AKS automatically apply matching `tolerations`_ 
+to pods requesting GPU resources; tolerations are applied by means of Kubernetes's `ExtendedResourceToleration`_ `admission controller`_.
```
Comment (richardliaw) [inline doc/source/cluster/kubernetes.rst:344]:
```suggestion
Managed Kubernetes services like GKE, EKS, and AKS automatically apply matching `tolerations`_ 
to pods requesting GPU resources. Tolerations are applied by means of Kubernetes's `ExtendedResourceToleration`_ `admission controller`_. 
```

File: doc/source/cluster/kubernetes.rst [hunk 4]
```diff
@@ -292,6 +297,79 @@ To delete a running Ray cluster, you can run the following command:
...
+     key: nvidia.com/gpu
+     operator: Exists
+   ...
+   containers:
+   - name: ray-node
+     image: rayproject/ray:nightly-gpu
+     ...
+
+Further reference and discussion
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (richardliaw) [inline doc/source/cluster/kubernetes.rst:366]:
```suggestion
Further reference and discussion
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: doc/source/cluster/kubernetes.rst [hunk 5]
```diff
@@ -292,6 +297,79 @@ To delete a running Ray cluster, you can run the following command:
...
+You can read about Kubernetes device plugins `here <https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/>`__,
+about Kubernetes GPU plugins `here <https://kubernetes.io/docs/tasks/manage-gpus/scheduling-gpus>`__,
+and about Nvidia's GPU plugin for Kubernetes `here <https://github.com/NVIDIA/k8s-device-plugin>`__.
```
Comment (richardliaw) [inline doc/source/cluster/kubernetes.rst:369]:
```suggestion
Read about Kubernetes device plugins `here <https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/>`__,
about Kubernetes GPU plugins `here <https://kubernetes.io/docs/tasks/manage-gpus/scheduling-gpus>`__,
and about Nvidia's GPU plugin for Kubernetes `here <https://github.com/NVIDIA/k8s-device-plugin>`__.
```

**Comment by DmitriGekhtman** [comment]
Suggested changes committed -- thanks!

File: doc/source/cluster/kubernetes.rst [hunk 6]
```diff
@@ -292,6 +297,79 @@ To delete a running Ray cluster, you can run the following command:
 
   kubectl delete -f ray/doc/kubernetes/ray-cluster.yaml
 
+.. _k8s-gpus:
+
+Using GPUs
+----------
+
+Using GPUs on Kubernetes requires some setup.
+Here's where to find the relevant documentation for GPU usage on `GKE`_, for `EKS`_, and for `AKS`_.
```
Comment (richardliaw) [inline doc/source/cluster/kubernetes.rst:306]:
```suggestion
To use GPUs on Kubernetes, you will need to configure both your GKE setup and add additional values to your Ray cluster configuration.

For relevant documentation for GPU usage on different clouds, see instructions for `GKE`_, for `EKS`_, and for `AKS`_.
```

**Review by richardliaw — APPROVED** [review]
This looks great! I don't mind if the nits are not applied. Thanks a bunch for doing this!

File: doc/source/cluster/kubernetes.rst (hunk 6 above)
Comment (DmitriGekhtman) [reply doc/source/cluster/kubernetes.rst]: Ah actually I guess it should say "Kubernetes setup" rather "GKE setup" (we should probably be cloud-agnostic)

**Comment by DmitriGekhtman** [comment]
Nits are great. Nits applied. Good to merge, I think.

**Pull request merged**